File: SerpentineApi/SerpentineApi/Responses/MessageResponse.cs
namespace SerpentineApi.Responses;

public class MessageResponse : BaseResponse
{
    public string Content { get; set; } = null!;
    public Ulid? SenderId { get; set; }
    public Ulid GroupId { get; set; }
    public Ulid? ParentId { get; set; }
    public bool IsNotification { get; set; } = false;
    public string? SenderName { get; set; }
    
    public string? SenderProfilePictureUrl { get; set; }
    
    public string? SenderUsername { get; set; }
    
    public string? ParentContent { get; set; }
    
    public Ulid? ChannelId { get; set; }
    
    public string? GroupName { get; set; }
    
    public string? ChannelName {get; set;}
    
    
}

File: serpentineclient/src/models/responses/message-response.ts
import {BaseResponse} from "@/models/responses/base-response.ts";
import {UserResponse} from "@/models/responses/user-response.ts";

export class MessageResponse extends BaseResponse {

    content: string = "";
    senderId?: string= ""; 
    groupId: string= "";   
    parentId?: string= ""; 
    isNotification: boolean = false;
    senderName?: string = "";
    senderProfilePictureUrl?: string = "";
    senderUsername?: string = "";
    parentContent?: string = "";
    groupName?: string = "";
    channelName?: string = "";
    channelId?: string = "";
    
    
    //front-end-only properties
    isNewAndUnread: boolean = false;//this property is set when the message bubble gets into the viewport for the first time

 



    constructor(init?: Partial<MessageResponse>) {
        super(init);
        Object.assign(this, init);

        
    }

   



}